insights/parsers/md5check.py

```diff
@@ -7,61 +7,84 @@
 The name and md5 checksums of specified files are stored as lists. Each
 entry will be processed by the shared parser.
 
-This MD5CheckSum class provide the following properties:
+This MD5CheckSum class provide the following attributes:
 
-* ``filename``: the file name which derived from the output.
-* ``md5sum``: the checksums for the file.
+* ``files``: the files which derived from the output.
+* ``checksums``: the checksums for the files.
 """
+
 from .. import parser, CommandParser
 from insights.specs import Specs
+from insights.parsers import ParseException
 
 
-@parser(Specs.prelink_orig_md5)
 class MD5CheckSum(CommandParser):
     """
-    Class to parse ``prelink -y --md5`` or ``md5sum`` command information.
+    Common class to parse ``prelink -y --md5`` or ``md5sum`` command information.
 
     The output of these two commands contains two fields, the first is the
     md5 checksum and the second is the file name.
 
     Especially, we add /dev/null as one of the specified file to be checked,
     as ``md5sum`` command can read from stdin without feeding a file.
 
-    Attributes::
-
-        filename (str): File name.
-        md5sum (str): MD5 checksums of that file.
-
     Sample input for ``prelink -y --md5`` or ``md5sum`` command::
 
         d1e6613cfb62d3f111db7bdda39ac821  /usr/lib64/libsoftokn3.so
 
-    Examples::
-
-        >>> collection = shared[MD5CheckSum]
-        >>> len(collection) # All MD5 checksums in a list
+    Examples:
+        >>> md5info = shared[MD5CheckSum]
```

Comment:
`shared` is no longer used by insights-core 3.0, please update the example test correspondingly. See https://github.com/RedHatInsights/insights-core/blob/7b8762c9a377c45f4b022ee7e0c64fccff72293d/insights/parsers/ps.py#L181

Reply:
Got it.